blog/2015-02-26-Getting-Started-With_Docker-And-Wildfly.adoc

```diff
@@ -0,0 +1,195 @@
+Once this is run, the command will not only start up your container, but also launch the server
+in standalone mode, and connect a terminal to it so you can see the output. 
+
+Assuming you see the console output, and the server runs with no errors, then you're 
+generally set up and ready for the next step. 
+
+To terminate the container, you can simply press `Ctrl+C`. If, for some reason, 
+your image has frozen, and `Ctrl+C` isn't working, you can also run the following
+commands to kill it forcefully. 
+
+Let's see what containers are currently running:
+
+....
+[root@localhost jbds_wf_external]$ docker ps
+CONTAINER ID        IMAGE                  COMMAND                CREATED             STATUS              PORTS                NAMES
+a798d07859f8        jboss/wildfly:latest   "/opt/jboss/wildfly/   4 seconds ago       Up 3 seconds        8080/tcp, 9990/tcp   backstabbing_poitras   
+....
+
+To kill the container, we'd then execute `docker kill a798d07859f8`, replacing the given hash with your CONTAINER ID. 
+
+
+== What's Missing
+
+The default docker image is lacking a few things for our needs. 
+First, we'd generally like our Wildfly installation to expose our 
+management port if we
+require any management actions or JMX introspection. 
+We might also like SSH access to this machine, if we intend to treat it as a remote system
+and deploy our applications over SCP. 
+We might like to map in 
+some local folders to the running container, if we'd prefer a more quick-and-dirty solution. 
+We could also expose some of the ports from our container 
+as if they were ports on our localhost. 
+
+Depending on our usecase, we may do some or all of the above. 
+
+
+== Externally Managed Local Server With Deployment Folder Mapping
+
+For an externally managed server, where we won't be starting or 
+stopping it from Eclipse, there's a few small changes we can make
+in our command line to launch the image that can make the default
+image much more useful for us. 
+
+=== Viewing the Running Server's Welcome Page
+
+At each launch of our container, the container is granted an IP address. 
+If we were to run our container with the default command as shown above, 
+we'd need to constantly change the IP address of our Server Adapter in JBossTools. 
+To find out the IP of the container, we'd run the following:
+
+....
+[root@localhost docker]$ docker ps
+CONTAINER ID        IMAGE                  COMMAND                CREATED             STATUS              PORTS                                            NAMES
+f70149043400        jboss/wildfly:latest   "/opt/jboss/wildfly/   58 seconds ago      Up 58 seconds       0.0.0.0:8080->8080/tcp, 0.0.0.0:9990->9990/tcp   ecstatic_darwin     
+
+[root@localhost docker]$ docker inspect -f '{{ .NetworkSettings.IPAddress }}' f70149043400
+172.17.0.25
+....
+
+To see the Wildfly welcome page, we could open up any browser and browse to `http://172.17.0.25:8080`.  
```

Comment:
The 'docker inspect' command mentionned above also works on MacOSX, but it will return the IP of the container *inside* the boot2docker VM. This IP is *not* reachable from the host machine (eg: the browser). The IP to use to connect to a container would be provided by the 'boot2docker ip' command. The port to connect to the container is one of those returned by docker inspect -f "{{.NetworkSettings.Ports }}" containerId